pkg/porter/generateManifest.go

```diff
@@ -111,15 +113,83 @@ func (p *Porter) generateInternalManifest(ctx context.Context, opts BuildOptions
 			}
 		}
 		return e.SetValue(path+"digest", digest.String())
-
 	})
 	if err != nil {
 		return err
 	}
 
+	if p.IsFeatureEnabled(experimental.FlagDependenciesV2) {
+		if err = p.resolveDependencyDigest(ctx, e, regOpts); err != nil {
+			return err
+		}
+	}
+
 	return e.WriteFile(build.LOCAL_MANIFEST)
 }
 
+func (p *Porter) resolveDependencyDigest(ctx context.Context, e *yaml.Editor, opts cnabtooci.RegistryOptions) error {
```

Comment:
Could this be reworked to leverage existing [dependency resolution](https://github.com/getporter/porter/blob/main/pkg/manifest/manifest.go#L868) logic?  - moving this into "validate" logic may help us reduce LOC and improve efficiency :)